### PR TITLE
Add useful rpc functions to the node interface

### PR DESCRIPTION
**Pull request by mouadino**
It's useful to get information about processes running by calling the node
rpc interface. Example when it can be useful is lymph-chaos (
https://github.com/mouadino/lymph-chaos)

### CODE REVIEW

File: lymph/services/node.py [hunk 1]
```diff
@@ -136,3 +141,19 @@ def _get_metrics(self):
...
+    def get_processes(self, service_type=None):
+        result = []
+        for proc in self.processes:
+            if not service_type or proc.service_type == service_type:
+                result.append(proc)
+        return result
+
+    def restart(self, service_type=None):
+        for proc in self.get_processes(service_type):
+            proc.stop()
```
Comment (mamachanko) [inline lymph/services/node.py]: why's this restarting the processes?

File: lymph/services/node.py [hunk 2]
```diff
@@ -136,3 +141,19 @@ def _get_metrics(self):
         for process in self.processes:
             for metric in process._get_metrics():
                 yield metric
+
+    @rpc()
+    def get_services(self):
+        return map(operator.itemgetter(0), self._services)
```
Comment (emulbreh) [inline lymph/services/node.py]: list comprehension please :)

File: lymph/services/node.py [hunk 3]
```diff
@@ -59,6 +61,12 @@ def _get_metrics(self):
         except psutil.NoSuchProcess:
             pass
 
+    def __lymph_dump__(self):
```
Comment (emulbreh) [inline lymph/services/node.py]: single underscore

File: lymph/services/node.py (hunk 1 above)
Comment (emulbreh) [inline lymph/services/node.py]: because the node restarts processes when they die

**Comment by mouadino** [comment]
Fixed.

**Comment by mouadino** [comment]
Not relevant anymore

**Pull request closed without merging**